arrocke/gloss-translation
# chore: Add Docker support

## What has changed



## Connected Issues


closes #258 

## QA Steps


1. If you have an existing VM, then halt and destroy it using `vagrant halt` followed by `vagrant destroy`.
1. Complete all steps in the `Prerequisites` section in file `doc\vagrant.md`.
1. Complete steps 1 - 5 in the `Setup Steps` section in file `doc\vagrant.md`.
1. Follow step 7 for provider `VirtualBox`.
1. Review and complete the `Post-Installation Notes` steps as required, then perform the 4 QA steps below:
    1. Make sure you can access database using pgAdmin.
    1. Make sure you can ssh to VM.
    1. Start API server in VM and make sure you can access via host web browser.
    1. Start Web app in VM and make sure you can access via host web browser.
1. Once QA steps of VirtualBox VM are completed, halt and destroy VM by following step 1 above.
1. Follow step 7 of `doc\vagrant.md` using provider `Docker`.
1. Follow `Post-Installation Notes` and perform the 4 QA steps as described in step 5 above.

## Post-Deployment



- [ x] Nothing required

## Issue:
During development, I discovered that due to a bug introduced in Vagrant v2.4.0, `vagrant up --provider vagrant` command aborts with an error during creation of a docker container. As a result, I had to develop and test using Vagrant v2.3.7. This new bug is to be fixed in v2.4.1, which is yet to be released. A note about this issue has been included in the updated `vagrant.md` document.
